src/test/cloudWatchLogs/document/logStreamDocumentProvider.test.ts

```diff
@@ -7,34 +7,38 @@ import * as assert from 'assert'
 import * as vscode from 'vscode'
 import { CloudWatchLogsSettings } from '../../../cloudWatchLogs/cloudWatchLogsUtils'
 import { LogStreamDocumentProvider } from '../../../cloudWatchLogs/document/logStreamDocumentProvider'
-import { LogStreamRegistry, CloudWatchLogStreamData } from '../../../cloudWatchLogs/registry/logStreamRegistry'
+import {
+    LogStreamRegistry,
+    CloudWatchLogsData,
+    getLogEventsFromUriComponents,
+} from '../../../cloudWatchLogs/registry/logStreamRegistry'
 import { Settings } from '../../../shared/settings'
 
 describe('LogStreamDocumentProvider', function () {
-    let registry: LogStreamRegistry
-    let map: Map<string, CloudWatchLogStreamData>
-    let provider: LogStreamDocumentProvider
+    const map = new Map<string, CloudWatchLogsData>()
 
     const config = new Settings(vscode.ConfigurationTarget.Workspace)
+    const registry = new LogStreamRegistry(new CloudWatchLogsSettings(config), map)
 
     const registeredUri = vscode.Uri.parse('has:This')
     // TODO: Make this less flaky when we add manual timestamp controls.
     const message = "i'm just putting something here because it's a friday"
-    const stream: CloudWatchLogStreamData = {
+    const stream: CloudWatchLogsData = {
         data: [
             {
                 message,
             },
         ],
+        parameters: { streamName: 'stream' },
+        logGroupInfo: {
+            groupName: 'group',
+            regionName: 'region',
+        },
+        retrieveLogsFunction: getLogEventsFromUriComponents,
         busy: false,
     }
-
-    beforeEach(function () {
-        map = new Map<string, CloudWatchLogStreamData>()
-        map.set(registeredUri.path, stream)
-        registry = new LogStreamRegistry(new CloudWatchLogsSettings(config), map)
-        provider = new LogStreamDocumentProvider(registry)
-    })
+    const provider = new LogStreamDocumentProvider(registry)
+    map.set(registeredUri.path, stream)
```

Comment:
initializing non-trivial things in describe() scope can have weird bugs. Probably should be in beforeEach() or just before() if it resetting before each test is not needed.